### PR TITLE
fix documentation workflow and added script.py for typedoc MPX compatible

### PR DESCRIPTION
**Pull request by dhiraj0911**
**What kind of change does this PR introduce?**

- Bugfix for the automated documentation GitHub action PUSH workflow.
- Standardizes package manager to `npm` for both `talawa-api` and `talawa-admin`.
- Adds a Python script (`script.py`) to ensure Markdown files are MPX compatible.

**Issue Number:** #1279 

---

**Did you add tests for your changes?**

- No new tests added. The changes pertain to the GitHub Actions workflow and documentation updates.
- Manual testing was conducted to ensure functionality and MPX compatibility.

---

**If relevant, did you update the documentation?**

- Documentation updated to reflect the shift from npm to yarn and the integration of `script.py`.
- Details on workflow changes and MPX compatibility considerations with Docusaurus v3 included.

---

**Summary**

- Addresses issues with GitHub action for automated documentation generation.
- Resolves inconsistencies in package managers and non-compliance with Docusaurus v3's MPX standards.
- Includes transition to yarn and introduction of `script.py` for Typedoc-generated Markdown files.

---

**Does this PR introduce a breaking change?**

- No breaking changes introduced.
- Focuses on resolving workflow issues and enhancing compatibility, not affecting core software functionality.

---

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)?**

- Yes, the contributing guidelines have been followed.

### CODE REVIEW

File: .github/workflows/script.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        required=True,
+        help="Directory containing Markdown files to process."
+    )
+
+    args = parser.parse_args()
+
+    # Process each Markdown file in the directory
+    for root, _, files in os.walk(args.directory):
+        for file in files:
+            if file.endswith(".md"):
```
Comment (palisadoes) [inline .github/workflows/script.py:90]: Make this case insensitive. Do this for the other PR too

File: .github/workflows/script.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python3
+# -*- coding: UTF-8 -*-
```
Comment (palisadoes) [inline .github/workflows/script.py:2]: Change the name of the script to be more self-explanatory. Do this for the other PR too

File: .github/workflows/script.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    Returns:
+        None, writes the processed content back to the file.
+    """
+    with open(filepath, 'r', encoding='utf-8') as file:
+        content = file.read()
+
+    # Escape MPX characters
+    new_content = escape_mpx_characters(content)
+
+    # Write the processed content back to the file
```
Comment (palisadoes) [inline .github/workflows/script.py:60]: Only do this if there is a change. Do this for the other PR too.

File: .github/workflows/push.yml [hunk 4]
```diff
@@ -91,9 +91,13 @@ jobs:
...
-          yarn global add typedoc
-          yarn add typedoc-plugin-markdown
-          yarn typedoc --entryPoints src/components src/screens --out talawa-admin-docs --plugin typedoc-plugin-markdown --theme markdown --entryPointStrategy expand --exclude "**/*.test.ts" --exclude "**/*.css"
+          npm install --global typedoc
+          npm install typedoc-plugin-markdown
+          npm install --save-dev @types/node
+          npx typedoc --entryPoints src/components src/screens --out talawa-admin-docs --plugin typedoc-plugin-markdown --theme markdown --entryPointStrategy expand --exclude "**/*.test.ts" --exclude "**/*.css"
+          
+      - name: Make Markdown Files MPX Compatible
+        run: python ./script.py --directory talawa-admin-docs
```
Comment (palisadoes) [inline .github/workflows/push.yml:100]: Update the name accordingly. Do this in the other PR too

**Review by palisadoes — CHANGES_REQUESTED** [review]
See comments

**Comment by dhiraj0911** [comment]
Made necessary changes

**Review by palisadoes — CHANGES_REQUESTED** [review]
1. This works if there is only one update to the PR.
2. What happens if a person submits a PR and then does multiple commits so that this script is run multiple times? It would add an excessive number of `\` characters to the files.
3. You will need to ensure that `\<`, `\>`, `\{`, and `\}` are not replaced with `\\<`, `\\>`, `\\{`, and `\\}`

**Pull request closed without merging**